randovania/game_connection/executor/am2r_executor.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import asyncio
+import dataclasses
+import logging
+import struct
+import typing
+from asyncio import StreamReader, StreamWriter
+from enum import IntEnum
+
+from PySide6.QtCore import QObject, Signal
+
+
+class AM2RConnectionException(Exception):
+    pass
+
+
+@dataclasses.dataclass()
+class AM2RSocketHolder:
+    reader: StreamReader
+    writer: StreamWriter
+    api_version: int
+    buffer_size: int
+    request_number: int
+
+
+class PacketType(IntEnum):
+    PACKET_HANDSHAKE = b"1"
+    PACKET_VERSION_AND_UUID = b"2"
+    PACKET_LOG_MESSAGE = b"3"
+    PACKET_NEW_INVENTORY = b"5"
+    PACKET_COLLECTED_INDICES = b"6"
+    PACKET_RECEIVED_PICKUPS = b"7"
+    PACKET_GAME_STATE = b"8"
+    PACKET_DISPLAY_MESSAGE = b"9"
+    PACKET_MALFORMED = b"10"
+
+
+class ClientInterests(IntEnum):
+    MULTIWORLD = b"2"
```

Comment:
Dunno if you really need this.
I used it to tell the game to send logs and/or multiworld packets in dread. Such that you can easily debug and get print logs from console.

Reply:
I technically don't need it, but it's nice. 
a) It makes it a little less likely that something else somehow connects to the game. 
b) I left this open for the future, cause I did toy around with the idea of having remote logging, or having other remote stuff.